src/main/kotlin/frc/robot/autonomous/AlignmentCommands.kt

```diff
@@ -0,0 +1,58 @@
+package frc.robot.autonomous
+
+import edu.wpi.first.math.geometry.Pose2d
+import edu.wpi.first.units.Units
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.Commands.defer
+import edu.wpi.first.wpilibj2.command.Commands.runOnce
+import edu.wpi.first.wpilibj2.command.button.Trigger
+import frc.robot.lib.distanceFromPoint
+import frc.robot.subsystems.outtakeCoral
+import frc.robot.swerveDrive
+import org.littletonrobotics.junction.AutoLogOutput
+import org.littletonrobotics.junction.Logger
+
+@AutoLogOutput(key = "AutoAlignment/IsAligning")
+var isAligning = Trigger { alignCommand().isScheduled }
+
+val pose: Pose2d
+    get() = swerveDrive.pose
+
+fun alignCommand(): Command {
+    return defer(
+        {
+            runOnce({ setGoal(selectedScorePose.invoke()) })
+                .andThen(
+                    swerveDrive.run {
+                        swerveDrive.normalRunVelocity(getSpeed(pose).invoke())
+                    }
+                )
+        },
+        setOf(swerveDrive)
+    )
+}
+
+@AutoLogOutput(key = "AutoAlignment/AtAlignmentSetpoint")
+private val atAlignmentSetpoint =
+    Trigger { atGoal.asBoolean && isAligning.asBoolean }.onTrue(outtakeCoral())
```

Comment:
```suggestion
    isAligningTrigger.and(atGoal).onTrue(outtakeCoral())
```
:)
I also changed the order. looks better.

Reply:
42e76f753eae9e4d362d1f3db9b9bcbb1022c538